lib/vault.ts

```diff
@@ -157,19 +161,30 @@ export async function listAllSubmissions(
         id: formID,
       },
       "ListResponses",
-      `List all responses for form ${formID}`
+      `List all responses ${status ? `of status ${status} ` : ""}for form ${formID}`
     );
 
-    const numOfUnprocessedSubmissions = accumulatedResponses.filter((submission) =>
-      [VaultStatus.NEW, VaultStatus.DOWNLOADED, VaultStatus.PROBLEM].includes(submission.status)
-    ).length;
-
-    await numberOfUnprocessedSubmissionsCachePut(formID, numOfUnprocessedSubmissions);
+    if (!status) {
+      // Only update value when we are not filtering by status
+      // Will need to rework logic once filtering is implemented on client side
+      const numOfUnprocessedSubmissions = accumulatedResponses.filter((submission) =>
+        [VaultStatus.NEW, VaultStatus.DOWNLOADED, VaultStatus.PROBLEM].includes(submission.status)
```

Comment:
Would adding `VaultStatus.CONFIRMED` be helpful also?

Reply:
`VaultStatus.Confirmed` would not be an unprocessed Submission.  This function is the existing one that tries to calculate the number of unconfirmed responses.
The logic of this will all be rewritten once we implement the filtering and pagination in the client side.